lib/source/file.js

```diff
@@ -0,0 +1,167 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const fs = require('fs');
+
+class FileParser {
+  constructor() {
+    this.properties = Object.create(null);
+  }
+
+  /**
+   * Parse the file contents
+   * @param {Buffer} data
+   */
+  update(data) {
+    let properties = Object.create(null);
```

Comment:
Yeah, I realized it doesn't actually matter here after I made the comment. We're setting `this.properties` to the properties of `properties` so it's always going to be `undefined` if parsing fails.